docs/high-avaliability/prometheus-HA.md

```diff
@@ -1,12 +1,65 @@
 
-Promscale supports running Prometheus in High Availability (HA). This means that more than one Prometheus instances can run in parallel. Promscale implements leader election to support Prometheus HA mode through pg_advisory_lock.
+Promscale is a stateless service, thus it can run safely with multiple
+replicas. A load-balancer can simply route Promscale requests to any replica.
 
-## Leader-election method via pg_advisory_lock
+# Using Promscale with Prometheus deployed in High Availability (HA) mode
 
-PostgreSQL provides a [pg_advisory_lock](https://www.postgresql.org/docs/current/explicit-locking.html#ADVISORY-LOCKS) that locks a application-defined resource based on an ID. Promscale makes use of this lock when running multiple promescale instances in parallel in order to choose one leader.
+Promscale supports running Prometheus in High Availability (HA). This means
+that you can have multiple Prometheus instances running as "HA
+pairs". Such "HA pairs" (what we will call clusters from now on) are simply two
+or more identical Prometheus instances running on different
+machines/containers. These instances scrape the same targets and should thus
+have very similar data (the differences occur because scrapes happen at slightly
+different times). You can think of a **cluster** as a group of similar
+Prometheus instances that are all scraping the same targets and **replicas**
+as individual giPrometheus instance in that group.
 
-![Promscale architecture for Prometheus in HA](https://github.com/Harkishen-Singh/promscale/blob/docs_HA_promscale/docs/assets/promscale-HA-arch.png?raw=true)
+Promscale supports de-duplicating data coming from such HA clusters upon
+ingest. It does this by electing a leader Prometheus instance and only
+ingesting data from that leader. If the leader Prometheus stops sending data
+to Promscale for any reason, Promscale will fail-over to the other Prometheus
+instance.
+
+Promscale supports having multiple clusters sending data to the same set of
+Promscale instances. It will elect one leader per cluster.
+
+## Prometheus leader-election via external labels
+
+![Promscale architecture for Prometheus in HA using external labels](https://raw.githubusercontent.com/timescale/promscale/master/docs/high-avaliability/new_ha_system.png)
+
+To set up Promscale to process data from Prometheus running in HA mode,
+Prometheus must be configured to communicate which cluster it belongs to
+using external labels. In particular, each Prometheus should send a `cluster`
+label to indicate which cluster it's in and a `__replica__` label that
+provides a unique identifier for the Prometheus instance with the cluster.
+Two Prometheus instances running as part of a HA cluster, MUST send the same
+`cluster` label and different `__replica__` labels.
+
+In Kubernetes environments, it is often useful to set the `__replica__`
+label to the pod name. When using the Prometheus Operator this can be done
```

Comment:
```suggestion
label to the pod name. When using the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator#prometheus-operator) this can be done
```

Reply:
Maybe it's good to reference the Prometheus Operator repo link here. So users understand what we are referring to here.